docs/notebooks/Common_Gotchas_in_JAX.ipynb

```diff
@@ -568,6 +568,209 @@
     "Note also that, as the two behaviors described above are not inverses of each other, reverse-mode automatic differentiation (which turns index updates into index retrievals and vice versa) [will not preserve the semantics of out of bounds indexing](https://github.com/google/jax/issues/5760). Thus it may be a good idea to think of out-of-bounds indexing in JAX as a case of [undefined behavior](https://en.wikipedia.org/wiki/Undefined_behavior)."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "LwB07Kx5sgHu"
+   },
+   "source": [
+    "## 🔪 Non-array inputs\n",
+    "\n",
+    "NumPy is generally happy accepting Python lists or tuples as inputs to its API functions:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {
+    "id": "sErQES14sjCG",
+    "outputId": "6bc29168-624a-4d51-eef1-220aeaf49985"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "6"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {
+      "tags": []
+     },
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "np.sum([1, 2, 3])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ZJ1Wt1bTtrSA"
+   },
+   "source": [
+    "JAX departs from this, generally returning a helpful error:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "metadata": {
+    "id": "DFEGcENSsmEc",
+    "outputId": "86105261-0aec-41e0-c8a6-16eec437e2a8"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "TypeError: sum requires ndarray or scalar arguments, got <class 'list'> at position 0.\n"
+     ]
+    }
+   ],
+   "source": [
+    "try:\n",
+    "  jnp.sum([1, 2, 3])\n",
+    "except TypeError as e:\n",
+    "  print(f\"TypeError: {e}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "QPliLUZztxJt"
+   },
+   "source": [
+    "This is a deliberate design choice, because passing lists or tuples to traced functions can lead to silent performance degradation that might otherwise be difficult to detect.\n",
+    "\n",
+    "For example, consider this the following permissive version of `jnp.sum` that allows list inputs:"
```

Comment:
Nit:

```suggestion
    "For example, consider the following permissive version of `jnp.sum` that allows list inputs:"
```

or maybe I misread this @jakevdp

Reply:
Done!